<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CharsetEditor</class>
 <widget class="QWidget" name="CharsetEditor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>595</width>
    <height>494</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Character Set Editor</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QRadioButton" name="useExistingRadio">
     <property name="text">
      <string>Use Existing CHR File</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="useExistingGroup">
     <property name="title">
      <string/>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_5">
      <item>
       <widget class="QLabel" name="label_2">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Character Set File:</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QLineEdit" name="chrFileEdit"/>
        </item>
        <item>
         <widget class="QPushButton" name="chrFileBrowse">
          <property name="text">
           <string>Browse...</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QLabel" name="maxPassLenLabel">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Max Password Length</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="maxPassLenEdit">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>32</width>
            <height>0</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>32</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="numCharsLabel">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Number Of Characters</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="numCharsEdit">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>32</width>
            <height>0</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>32</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QLabel" name="invalidCharsetLabel">
        <property name="text">
         <string>Character set file is not valid!</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="maskModeCheckBox">
        <property name="text">
         <string>Accelerate With Mask Mode</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="0">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Mask:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Mask Encoding:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLineEdit" name="maskEdit"/>
        </item>
        <item row="1" column="1">
         <widget class="QComboBox" name="maskEncodingCombo">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QRadioButton" name="createNewRadio">
     <property name="text">
      <string>Create New CHR File</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="createNewGroup">
     <property name="title">
      <string/>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>A character set can be defined by a set of input characters, a set of dictionary files and/or a set of potfiles from an external run of John The Ripper. Using dictionary or potfiles will consider letter and pairing frequency and generate a faster character set.</string>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_7">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_9">
          <item>
           <widget class="QCheckBox" name="useCharsCheckbox">
            <property name="text">
             <string>Use Input Characters</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="inputCharactersEdit"/>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_10">
          <item>
           <widget class="QLabel" name="inputCharactersEncodingLabel">
            <property name="text">
             <string>Input Characters Encoding:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="inputCharactersEncodingCombo">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_8">
          <item>
           <widget class="QCheckBox" name="usePotCheckbox">
            <property name="text">
             <string>Use Input John .pot Files</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <item>
             <widget class="QListWidget" name="potList">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Ignored">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_4">
              <item>
               <widget class="QToolButton" name="plusButtonPot">
                <property name="text">
                 <string/>
                </property>
                <property name="icon">
                 <iconset resource="../../lc7/resources/lc7.qrc">
                  <normaloff>:/lc7/plus.png</normaloff>:/lc7/plus.png</iconset>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QToolButton" name="minusButtonPot">
                <property name="text">
                 <string/>
                </property>
                <property name="icon">
                 <iconset resource="../../lc7/resources/lc7.qrc">
                  <normaloff>:/lc7/minus.png</normaloff>:/lc7/minus.png</iconset>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="verticalSpacer_2">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeType">
                 <enum>QSizePolicy::Expanding</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>0</width>
                  <height>0</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_6">
          <item>
           <widget class="QCheckBox" name="useDictCheckbox">
            <property name="text">
             <string>Use Input Dictionary Files</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="QListWidget" name="dictionaryList">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Ignored">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_3">
              <item>
               <widget class="QToolButton" name="plusButtonDict">
                <property name="text">
                 <string/>
                </property>
                <property name="icon">
                 <iconset resource="../../lc7/resources/lc7.qrc">
                  <normaloff>:/lc7/plus.png</normaloff>:/lc7/plus.png</iconset>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QToolButton" name="minusButtonDict">
                <property name="text">
                 <string>...</string>
                </property>
                <property name="icon">
                 <iconset resource="../../lc7/resources/lc7.qrc">
                  <normaloff>:/lc7/minus.png</normaloff>:/lc7/minus.png</iconset>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="verticalSpacer">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeType">
                 <enum>QSizePolicy::Expanding</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>0</width>
                  <height>0</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <item>
         <widget class="QPushButton" name="createButton">
          <property name="text">
           <string>Create</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../../lc7/resources/lc7.qrc"/>
 </resources>
 <connections/>
</ui>
